README.md

```diff
@@ -70,3 +70,86 @@ In case you need to specify multiple values for option, specify it multiple time
 ```
 > dotnet migrate-2017 migrate -t net40 -t net45
 ```
+
+## Use as a NuGet library from your own code
+
+For additional control of the project migration process, you can use the NuGet packages directly from your code.
+
+Add the `Project2015To2017.Migrate2017.Library` package to your project e.g.
+
+```
+dotnet add package Project2015To2017.Migrate2017.Library
+```
+
+Then, to apply the default project migration:
+
+```c#
+using Project2015To2017;
+using Project2015To2017.Migrate2017;
+using Project2015To2017.Writing;
+
+//We use Serilog, but you can use any logging provider
+using Serilog;
+using Serilog.Extensions.Logging;
+
+namespace Acme.ProjectMigration
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var logger = new LoggerConfiguration()
+                                .Enrich.FromLogContext()
+                                .MinimumLevel.Debug()
+                                .WriteTo.Console()
+                                .CreateLogger();
+
+            var genericLogger = new SerilogLoggerProvider(logger).CreateLogger(nameof(Serilog));
+
+            var facility = new MigrationFacility(genericLogger);
+
+            facility.ExecuteMigrate(
+                    new[] { @"c:\full-path-to-solution-or-project-file.sln" },
+                    Vs15TransformationSet.Instance, //the default set of project file transformations
+                    new ConversionOptions(), //control over things like target framework and AssemblyInfo treatment
```

Comment:
Feels like we should indicate that all parameters (other than the first) are optional. Probably change API contract for that (make all optional with default value to null, inside implementation check if parameter is null and use default). Non-blocking.

Reply:
Thanks, I'll leave as is for now but sounds like a good idea